Dockerfile

```diff
@@ -34,11 +34,13 @@ FROM --platform=$BUILDPLATFORM golang:${GO_VERSION}-alpine${ALPINE_VERSION} AS g
 
 # git stage is used for checking out remote repository sources
 FROM --platform=$BUILDPLATFORM alpine:${ALPINE_VERSION} AS git
-RUN apk add --no-cache git
+RUN apk add --no-cache git && \
+  git config --global advice.detachedHead false
```

Comment:
What's the implication of this?

Reply:
If we switch to detached HEAD in git, we will get the following message

```
Note: switching to 'v1.1.13'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false
```

The current `Dockerfile` avoids this output by adding the `-q` option to the `checkout` command to suppress all checkout messages, but if we run `git clone` with the `-b` option, these messages will be output even with `-q`.

To prevent this, simply set `advice.detachedHead` to `false` as shown in the message, so we need to execute `git config --global advice.detachedHead false` beforehand, or add the `-c advice.detachedHead=false` option to each command.

Since there are many `git clone` and `git checkout` commands in the `Dockerfile`, I chose to use `git config` to simplify each command.